### ISSUE TITLE
The *Messages* Buffer

### ISSUE THREAD

**Issue by Remillard** [SystemCrafters/crafted-emacs#418]
Is there something in Crafted Emacs that removes or kills the `*Messages*` buffer?  While trying to debug something with tree-sitter, I realized I didn't have the buffer in the list anymore.  In `init.el` I have a lot of `(message "-- string")` commands to help keep track of where I am in the setup if something occurs.  However I've been grepping through the whole of my `.emacs.d` directory and I can't find anything that explicitly says it turns off `*Messages*`.  There is a sort of `kill-buffer` in `crafted-startup-config.el` but it's not really directly related to `*Messages*`.

I can get it back via a few ways, but it seems strange that it just disappeared and I think Crafted is the most likely candidate since it fiddles with the startup screen.  But I don't know enough complicated Elisp to figure it out.

**Comment by sthesing** [comment]
I don't think that Crafted Emacs is the culprit, here. _If_ it is, it's certainly unintended.
A quick way to test that is to comment out the `(require 'crafted-startup-config)` in your config. It won't help you identify the exact source of the phenomenon, but it'll help to eliminate `crafted-startup-config` as a suspect.

**Comment by jeffbowman** [comment]
The line you are referring to has to do with what Emacs will do when you press `q` on the splash screen, in this case, it will kill the splash screen buffer.  

If you could share a link to your current configuration, perhaps one of us could look over it and help diagnose the problem.

**Comment by Remillard** [comment]
Oh sure, I keep it version controlled.  

https://github.com/Remillard/dotemacsd

I think it's reasonably organized, but as things are tinkered and I either decide they work for me or not, there's cruft in there.

**Comment by Remillard** [comment]
Ohhh, let me double check.  I don't think there's anything particularly sensitive there.  I did try to come up with a way to reference localized personal stuff, though might be some project names.  Not a big deal though.  I'll go make sure it's readable.  Sorry about that.

**Comment by jeffbowman** [comment]
Yep, can read it now.  I'll see if I can help.

**Comment by jeffbowman** [comment]
Sorry for the delay, work and stuff.  

Here is what I have found.  Starting Emacs and then pressing `C-x b` does not display the `*Messages*` buffer in the list in the minibuffer.  However, pressing `C-x C-b` does show the `*Messages*` buffer.   You have the former set to run `tabspaces-switch-to-buffer` which does not show the `*Messages*` buffer in the list.  You can still switch to it by typing the name.  Once you have switched to it, it shows as a "live" buffer in the tab or frame.  This seems to be by design.  If you always want the `*Messages*` buffer to show up in the list, you can customize the `tabspaces-include-buffers`.  The `crafted-workspaces-config` module customizes this to be the list `'(*scratch*)`, which is also the default.  

Does this help?

**Comment by Remillard** [comment]
I will definitely check this when I get back to work on Monday.  Thanks for looking into this.  I'll pop back and close after tweaking.

**Comment by Remillard** [comment]
Okay, I see your point about them not showing up in the quick matching buffer switch but they do show up in the list (which is bound to `bufler` for its organizational design).  And I do see the tabspaces setting.  Is this a list?  Generally I would be pretty satisfied if `*Messages*` was in the tabspaces list because well I mess things up on the regular so I like to check the output.  I looked at the variable definition, but didn't really give me much insight into how to alter it.  It doesn't appear to be an `alist` at least on the surface, so `add-to-alist` probably won't be the right thing.  I suppose I could just redefine it at some point in `init.el`.

**Comment by jeffbowman** [comment]
You can use `M-x customize-option RET tab-spaces-include-buffers RET` as one option.  Another option is to add `(add-to-list 'tab-spaces-include-buffers '*Messages*)` somewhere in you `init.el` **after** you `(require 'crafted-workspaces-config)`  or you can just replace the customization in your `init.el` file (again, **after** you require it), `(customize-set-variable 'tab-spaces-include-buffers '(*scratch* *Messages*))`

There are other ways, but these are (I think) the easiest.  Personally, I'd opt for the first one, but that's up to you.

**Comment by Remillard** [comment]
Well I tried a few of these.  Generally a fan of explicitly stating things, I put 
```
;;
;; Making sure *Messages* shows up as a live buffer rather than being
;; secondarily accessed.
;;
(add-to-list 'tabspaces-include-buffers "*Messages*")
```
at the end of `init.el`.  This didn't seem to work.  I checked the variable contents and it was still only `*scratch*`.

Next I changed that line to:

`(customize-set-variable 'tabspaces-include-buffers '(*scratch* *Messages*))`

inside `init.el`.  This seemed to correctly change the variable.  When executed at the end of the file, and checked after the launch of Emacs, the value of `tabspaces-include-buffers` is `(*scratch* *Messages*)`.

However, I still don't see it listed if I do a short buffer switch `C-x b [select from list]`.  I'm not sure what keeps it out of the list for `tabspaces-switch-to-buffer`.  Seems very strange to me.  But as you rightly point out it IS available in `buffer-menu` and (probably because of this) also available in `bufler-list` which I have bound over `C-x C-b`.  Also I solved my problem with tree-sitter so until the next snafu I'm in less need of checking error messages.  Just kind of curious what is actively keeping it off of the `tabspaces` list.  Thanks for all the help!
